components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+        gauge.labels(host, port, javaHome, javaVersion).set(System.currentTimeMillis());
+    }
+
+    @Override
+    public void serverDown(String host, String port, String javaVersion, String javaHome) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVER_UP);
+        gauge.labels(host, port, javaHome, javaVersion).set(0);
+    }
+
+    @Override
+    public void serviceUp(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(System.currentTimeMillis());
+        System.out.println(gauge.labels(serviceName, serviceType).get());
+        System.out.println(System.currentTimeMillis());
+    }
+
+    @Override
+    public void serviceDown(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(0);
+    }
+
+    @Override
+    public void initMetrics() {
+        PrometheusMetricCreator prometheusMetricCreator = new PrometheusMetricCreator();
+        prometheusMetricCreator.createProxyServiceMetric();
+        prometheusMetricCreator.createAPIServiceMetric();
+        prometheusMetricCreator.createInboundEndpointMetric();
+        prometheusMetricCreator.createProxyServiceErrorMetric();
+        prometheusMetricCreator.createApiErrorMetric();
+        prometheusMetricCreator.createInboundEndpointErrorMetric();
+
+        prometheusMetricCreator.createServerUpMetrics();
+        prometheusMetricCreator.createServiceUpMetrics();
+    }
+
+    enum SERVICE {
+        PROXY,
+        API,
+        INBOUND_ENDPOINT
+    }
+
+    /**
+     * Instrument metrics on server deployment.
+     *
+     * @param configs The map of configs defined in the deployment.toml file
+     */
+    private void createBuckets(Map<String, Object> configs) {
+        proxyLatencyBuckets = new double[]{0.19, 0.20, 0.25, 0.30, 0.35, 0.40, 0.50, 0.60, 1, 5};
+        apiLatencyBuckets = new double[]{0.19, 0.20, 0.25, 0.30, 0.35, 0.40, 0.50, 0.60, 1, 5};
+        inboundEndpointLatencyBuckets = new double[]{0.19, 0.20, 0.25, 0.30, 0.35, 0.40, 0.50, 0.60, 1, 5};
+
+        Object proxyConfigBuckets = configs.get(MetricConstants.PROMETHEUS_HANDLER + "." +
+                MetricConstants.PROXY_LATENCY_BUCKETS);
+        Object apiConfigBuckets = configs.get(MetricConstants.PROMETHEUS_HANDLER + "." +
+                MetricConstants.API_LATENCY_BUCKETS);
+        Object inboundEndpointConfigBuckets = configs.get(MetricConstants.PROMETHEUS_HANDLER + "." +
+                MetricConstants.INBOUND_ENDPOINT_LATENCY_BUCKETS);
+
+        if (null != proxyConfigBuckets) {
+            List<Object> list = Arrays.asList(proxyConfigBuckets);
+
+            int size = ((ArrayList) proxyConfigBuckets).size();
+            for (int i = 0; i < size; i++) {
+                proxyLatencyBuckets[i] = (double) ((ArrayList) list.get(0)).get(i);
```

Comment:
Shall we get the 0th element outside the loop and assign it to a variable to use it here.

Reply:
Fixed